scripts/createTsIconComponents.mjs

```diff
@@ -6,51 +6,54 @@
  *
  */
 
-import { readFile, readdir, writeFile } from 'fs/promises';
-import { join, dirname } from 'path';
-import { transform } from '@svgr/core';
+import { readFile, readdir, stat, writeFile } from "fs/promises";
+import { join, dirname } from "path";
+import { transform } from "@svgr/core";
 
 const __dirname = dirname(new URL(import.meta.url).pathname);
-const rootDir = join(__dirname, '..', 'packages', 'ndla-icons');
+const rootDir = join(__dirname, "..", "packages", "ndla-icons");
+const today = new Date(new Date().setHours(0, 0, 0, 0));
 
-const copyright = await readFile(join(__dirname, '..', 'COPYRIGHT'), 'utf-8');
+const copyright = await readFile(join(__dirname, "..", "COPYRIGHT"), "utf-8");
 
 const autoNotice = `// N.B! AUTOGENERATED FILE. DO NOT EDIT`;
 const imports = `import Icon, { Props } from '../Icon'`;
 
 const allFileNames = await readdir(`${rootDir}/svg`, { recursive: true });
-const svgFiles = allFileNames.filter((file) => file.endsWith('.svg'));
+const svgFiles = allFileNames.filter((file) => file.endsWith(".svg"));
 const allFiles = await Promise.all(
   svgFiles.map(async (filename) => {
     const filePath = `${rootDir}/svg/${filename}`;
-    const file = await readFile(filePath, 'utf8');
-    return { filename, file };
+    const { ctime, mtime } = await stat(filePath);
+    const file = await readFile(filePath, "utf8");
+    return { filename, file, ctime, mtime };
   }),
 );
-
-const filePromises = allFiles.map(async ({ filename, file }) => {
-  const [folderName, iconName] = filename.split('/');
-  const componentName = iconName.split('.')[0];
+const filePromises = allFiles.map(async ({ filename, file, ctime, mtime }) => {
+  if (mtime && mtime < today) return;
```

Comment:
Dette krever vel at fila må være oppdatert i dag, men det er uansett bedre enn det var før.

Reply:
Ja, det er for å unngå tukling med filer som ikke ska tukles med. Tanken er at det ska virke hvis vi oppdaterer et ikon vi har hatt lenge også.
Kan alternativt legge inn litt leeway, men har tenkt at scriptet kjøres når svg oppdateres / lages

Reply:
Har sjekka litt, og det ser ut til å være et par måter det kan gjøres på. Men det vil allikevel føre til endringer fra de årstallene som står der. F.eks. når jeg sjekker history på AlertCircle.svg, finner jeg git log på den til 2019, men fila vår sier 2021 før denne. Hvis det er noe vi tenker er greit, så kan jeg legge det til.